lib/puppet/provider/archive/wget.rb

```diff
@@ -0,0 +1,42 @@
+Puppet::Type.type(:archive).provide(:wget, :parent => :ruby ) do
+
+  commands   :wget => '/usr/bin/wget'
+  defaultfor :operatingsystem => :linux
```

Comment:
I wonder if it just used the default ruby provider when you were testing. I don't see a way to test for negate in defaultfor, only for confine in the puppet provider code.